### PR TITLE
Allow user to include Youtube channel in Signup

### LINKED ISSUE #2676: Allow YouTube channels to be added via the sign-up UI
In #2675 we're adding support for YouTube feed discovery.  We need to update the UI to allow a user to include their YouTube channel, along with their blog(s).

### PR DESCRIPTION
**Pull request by tcvan0707**
## Issue This PR Addresses

Fixes #2676 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Description

This PR will allow user to include their Youtube channel when registering their account in Telescope.

NOTE: This PR is not fully completed yet. I am still not able to get the feed from Youtube channel, and the `Youtube URL` does not get the `MuiInputLable-shrink`. Advices are welcome ^^

## Steps to test the PR

- run `pnpm services:start`
- go to `localhost:8000`
- click the Sign In button
- Use the test account to sign in in SAML: username: `user1`, password: `user1pass`
- If successful, your account is created and can be found at `http://localhost:8910/project/default/editor` under the `telescope_profiles` table

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/api/sso/src/middleware.js [hunk 1]
```diff
@@ -106,6 +106,7 @@ const createNewProfile = async (id, body) => {
     githubUsername,
     githubAvatarUrl,
     blogUrl,
+    youtubeUrl,
```
Comment (humphd) [inline src/api/sso/src/middleware.js:109]: I don't think we need to modify the schema of a user in the DB.  A YouTube channel is just another RSS feed that happens to be hosted on YouTube.

**Review by humphd — COMMENTED** [review]
I think this is more complicated than it needs to be.  We only need to make it clear that you can specify a YouTube channel as one of your feeds, and the rest should just work (i.e., it's a normal feed URL).

**Comment by DukeManh** [comment]
@tcvan0707, according to your comment, here are my ideas to implement this:

- Add another page to the signup flow for users' Twitch account, it has 3 pages currently, 1 for the general info, 1 for the blog URL, and 1 for the GitHub account.
- This page is skippable, unlike the required Blog URL or GitHub account. We are using Formik to validate inputs, to make this page skippable, add a rule in Formik that allows empty value of the  twitch channel
- When users enter the twitch channel name, make a request to our RSS-bridge service to get the feed-URL
- After the signup flow, you may end up with data like:
```js
  blogs: [
    { URL: '123', type: 'blog'},
    { URL: '456', type: 'twitch'},
  ]
```
- Send this data to SSO service which will then handle this data and add insert the twitch feed URL to the database.

**Comment by humphd** [comment]
@tcvan0707 why did we close this?

**Comment by tcvan0707** [comment]
> @tcvan0707 why did we close this?

I am working on changes, so I will close this one and open the new PR, or can we convert this one to the actual PR?

**Comment by TueeNguyen** [comment]
@tcvan0707 this is good for review right?

**Comment by tcvan0707** [comment]
> @tcvan0707 this is good for review right?

This one will be closed in the future via #3505

**Comment by TueeNguyen** [comment]
Ok, focus on the other one then

**Comment by tcvan0707** [comment]
Closed via #3505

**Pull request closed without merging**